rcon/squad_automod/automod.py

```diff
@@ -219,80 +222,59 @@ def get_punitions_to_apply(rcon, config: NoLeaderConfig) -> PunitionsToApply:
-                        punitions_to_apply.punish.append(
-                            APlayer(
-                                player=player["name"],
-                                team=team,
-                                squad=squad_name,
-                                role=player.get("role"),
-                                lvl=player.get("level"),
-                            )
-                        )
+                        punitions_to_apply.punish.append(a_player)
                         punitions_to_apply.add_squad_state(squad_name, squad)
                     if state != PunishStepState.go_to_next_step:
                         continue
 
                     state = should_kick_player(
-                        watch_status, config, team_view, team, squad_name, squad, player
+                        watch_status, config, team_view, squad_name, squad, player
                     )
                     if state == PunishStepState.apply:
-                        punitions_to_apply.kick.append(
-                            APlayer(
-                                player=player["name"],
-                                team=team,
-                                squad=squad_name,
-                                role=player.get("role"),
-                                lvl=player.get("level"),
-                            )
-                        )
+                        punitions_to_apply.kick.append(a_player)
                         punitions_to_apply.add_squad_state(squad_name, squad)
 
     return punitions_to_apply
 
 
-def _build_warning_str(punition_to_apply: PunitionsToApply, config: NoLeaderConfig, team: str):
-    to_apply = punition_to_apply.warning.get(team.lower(), [])
-    if not to_apply:
-        return ""
-    # Since we're going to broadcast anyways we also add the squads got warned already but are still without lead
-    to_apply += punition_to_apply.pending_warnings.get(team.lower(), [])
-    red = get_redis_client()
-    start = f"{team}: "
-    squads = []
-    for idx, squad_name in enumerate(to_apply):
-        with watch_state(red, team.lower(), squad_name) as squad_state:
-            nb = len(squad_state.warned)
-            total = config.number_of_warning
-            squads.append(f"{str(squad_name).upper()} {nb}/{total} /!\\")
-
-    return f"{start}{', '.join(squads)}"
-
-
-def get_warning_message(punition_to_apply: PunitionsToApply, config: NoLeaderConfig) -> Tuple[str, str]:
-    allies = _build_warning_str(punition_to_apply, config, "ALLIES")
-    axis = _build_warning_str(punition_to_apply, config, "AXIS")
-    return f"""{config.warn_message_header}
-{allies}
-{axis}
-{config.warn_message_footer}
-""", f"{allies} {axis}"
+def get_warning_message(config: NoLeaderConfig) -> str:
+    if config.warn_message_header != "" or config.warn_message_footer != "":
+        return f"""{config.warn_message_header}
+{config.warn_message_footer}"""
+    else:
+        return config.warning_message
```

Comment:
I think this is a bit confusing. Why don't we just keep the old parameter? We should actually show how many warning a player has left before he's going to get slapped, that's a good motivation. We can insert it between the header and footer

Reply:
I liked the idea of @cemathey. Instead of building this on our own, we could provide the user the possibility to write the whole message on their own. We could then provide them with variables they can use as placeholders, like the players name, the number of warnings the player got already, the warnings in total that are required to get to the next stage, and so on.

Reply:
I added the formatting option now. I think this way we give the community a much more flexible solution. What do you think? :) If this is something we like, we could also add this to the other messages as well (however, there are probably much more restrictions with regards to the length 🙈